packages/assets-controllers/src/TokenListController.ts

```diff
@@ -204,104 +204,110 @@ export class TokenListController extends StaticIntervalPollingController<TokenLi
+   * Consider using the new polling approach instead
    */
   stop() {
     this.stopPolling();
   }
 
   /**
-   * Prepare to discard this controller.
-   *
    * This stops any active polling.
+   * @deprecated This method is deprecated and will be removed in the future.
+   * Consider using the new polling approach instead
    */
   override destroy() {
     super.destroy();
     this.stopPolling();
   }
 
+  /**
+   * This stops any active polling intervals.
+   * @deprecated This method is deprecated and will be removed in the future.
+   * Consider using the new polling approach instead
+   */
   private stopPolling() {
     if (this.intervalId) {
       clearInterval(this.intervalId);
     }
   }
 
   /**
-   * Starts a new polling interval.
+   * Starts a new polling interval for a given chainId (this should be deprecated in favor of _executePoll)
+   * @deprecated This method is deprecated and will be removed in the future.
+   * Consider using the new polling approach instead
    */
-  async #startPolling(): Promise<void> {
-    await safelyExecute(() => this.fetchTokenList());
+  async #startDeprecatedPolling(): Promise<void> {
+    // renaming this to avoid collision with base class
+    await safelyExecute(() => this.fetchTokenList(this.chainId));
     // TODO: Either fix this lint violation or explain why it's necessary to ignore.
     // eslint-disable-next-line @typescript-eslint/no-misused-promises
     this.intervalId = setInterval(async () => {
-      await safelyExecute(() => this.fetchTokenList());
+      await safelyExecute(() => this.fetchTokenList(this.chainId));
     }, this.intervalDelay);
   }
 
   /**
-   * Fetching token list from the Token Service API.
+   * This starts a new polling loop for any given chain. Under the hood it is deduping polls
    *
    * @private
    * @param input - The input for the poll.
-   * @param input.networkClientId - The ID of the network client triggering the fetch.
+   * @param input.chainId - The chainId of the chain to trigger the fetch.
    * @returns A promise that resolves when this operation completes.
    */
-  async _executePoll({
-    networkClientId,
-  }: TokenListPollingInput): Promise<void> {
-    return this.fetchTokenList(networkClientId);
+  async _executePoll({ chainId }: TokenListPollingInput): Promise<void> {
+    return this.fetchTokenList(chainId);
   }
 
   /**
-   * Fetching token list from the Token Service API.
+   * Fetching token list from the Token Service API. This will fetch tokens across chains. It will update tokensChainsCache (scoped across chains), and also the tokenList (scoped for the selected chain)
    *
-   * @param networkClientId - The ID of the network client triggering the fetch.
+   * @param chainId - The chainId of the current chain triggering the fetch.
    */
-  async fetchTokenList(networkClientId?: NetworkClientId): Promise<void> {
+  async fetchTokenList(chainId: Hex): Promise<void> {
```

Comment:
What are your thoughts on continuing to take a network client ID until we remove the global network completely? My thought is that clients might find it easier to pass this instead of a chain ID. The selected network client ID can be easily obtained from NetworkController state as `selectedNetworkClientId` and accessed within a Redux selector, while the client must obtain the chain ID by either doing the same thing that's happening here (use the messenger to get the network client by its ID, then get the chain ID that corresponds to it), or if within a Redux selector, loop through the `networkConfigurationsByChainId` to find the network configuration that matches the `selectedNetworkClientId` and then get the `chainId` off of that.

Ultimately I guess it depends on what the clients are currently doing. If we already have a way of accessing the current chain ID in a more or les type-safe way, then no worries. So, maybe this is change is just fine, but I wanted to double-check?

Reply:
The move towards passing in the chain ID, rather than an networkClientId was an intentional one. You're right, it's one of the steps towards removing the global network selector entirely. Yes, the clients are now executing a single poll per `ChainId`.

I'll defer to Brian here for more on these design decisions, but the pattern with the other AssetsControllers are also moving towards this same way of passing chainID instead of networkClientId.

Reply:
We're going to fetch token lists across all chains rather than just the current chain, so from the perspective of token lists we already no longer care about the current network.   It'll just do `Object.keys(networkConfigurationsByChainId)` and fetch for each chain id.  So the chain id input is already more convenient.